packages/fxa-settings/src/pages/ResetPassword/container.tsx

```diff
@@ -0,0 +1,177 @@
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+
+// WIP PROTOTYPE
+// ❌ = Note for prototype review. Comment will be removed in implementation.
+
+export interface PasswordForgotSendCodeResponse {
+  passwordForgotSendCode: {
+    passwordForgotToken: string;
+  };
+}
+
+export type BeginResetPasswordHandler = (
+  email: string,
+  service?: MozServices
+) => Promise<void>;
+
+export interface BeginResetPasswordResult {
+  // ❌ 'data' is undefined when the mutation is loading and is 'null' when there
+  //  is an error
+  data: PasswordForgotSendCodeResponse | null | undefined;
+  loading: boolean;
+  error?: {
+    errno: number;
+    message: string;
+    ftlId: string;
+    retryAfterLocalized?: string;
+  };
+}
+
+export const BEGIN_RESET_PASSWORD_MUTATION = gql`
+  mutation passwordForgotSendCode($input: PasswordForgotSendCodeInput!) {
+    passwordForgotSendCode(input: $input) {
+      passwordForgotToken
+    }
+  }
+`;
+
+// ❌ Every page component that needs an API call or data validation will have a
+// corresponding container component to handle it.
+const ResetPasswordContainer = (_: RouteComponentProps) => {
+  // ❌ Instead of model.method in page components, e.g. `account.resetPassword`,
+  // queries and mutations will be executed in a page container component. This
+  // is possible by wrapping our app in `<ApolloProvider>` and allows:
+  // 1) Keeping the presentation layer light. "Heavy lifting" in our components is
+  // one reason we created the Account model (see pull/8138)
+  // 2) Better separation of our API/network layer from our data model. The
+  // Account model acts as both and has grown too large; having a per-page
+  // container makes it easy to see what data needs to be fetched and used and
+  // what processing occurs for that page. Keeping GQL in only this layer also
+  // makes it easier to refactor away from GQL if we ever desire to in the future
+  // 3) Using hooks instead of `apolloClient.query` or `apolloClient.mutate`
+  // which has several benefits, including attaching to the React lifecycle,
+  // easier polling, and more documentation
+  // 4) Easier testing. By never using `useQuery` or `useMutation` in our
+  // presentation layer, we don't need to use `MockedProvider`, which can require
+  // heavy mocking and setup, _except_ for in page container tests. This also
+  // provides more inversion of control.
+  const [beginResetPassword, beginResetPasswordMutationResult] =
+    useMutation<PasswordForgotSendCodeResponse>(BEGIN_RESET_PASSWORD_MUTATION);
+  const [beginResetPasswordError, setBeginResetPasswordError] =
+    useState<BeginResetPasswordResult['error']>(undefined);
+
+  // ❌ Validate all the things. We may prefer for the model to handle these
+  // variations for us, e.g. we have one model per page and we have it attempt
+  // to read from router state, then localStorage, then query params but,
+  // it may also be nice to be more explicit.
```

Comment:
@dschom I realize after pushing this comment that "router state, then localStorage, then query params" might not be the right order, but I think you could sway me that it's more ideal for there to be one model per page and we don't care to be explicit. We should know what values might be in localStorage, but I might need to think about how we can allow the model to decide that we're pulling the data from the right place. I'm not sure if there's some case where there's an `email` param, for example, that might be available in local storage, but that we want to use the value in the query param instead, or vice versa.

Reply:
Although the `@bind` directives is currently designed to use a single data source, it could be adapted to do what you are suggesting. For example, the `@bind` directive could indicate where to store to and retrieve the data from. The defaults order of looks ups could be router state, storage state, and finally query params, but there'd always be an option override this behavior. For example by default maybe it looks like this:

```
class Model {
  @IsString()
  @bind()
  public fooBar
```
But if we wanted to only have the value stored in reach router and session storage we could declare it this way
```
class Model {
  @IsString()
  @bind([router,session])
  public fooBar
```
(Note, that I am using the class-validator lib here and assuming it's compatible.)

Reply:
I like this idea. I'm thinking I'll punt the decision/exploration to FXA-7308 and make note of the decision in the 2nd ADR mentioned earlier (which will probably a separate task mostly for documentation's sake) after getting a PR up for FXA-7308.